server/cmwell-dc/src/main/scala/cmwell/dc/stream/Util.scala

```diff
@@ -97,4 +98,36 @@ object Util extends LazyLogging {
   def headersString(headers: Seq[HttpHeader]): String =
     headers.map(headerString).mkString("[", ",", "]")
 
+  def createInfotonDataTransformer(dcInfo: DcInfo): InfotonData => InfotonData = {
+    if (dcInfo.transformations.isEmpty) identity
+    else {
+      val transformations = dcInfo.transformations.toList
+      infotonData => {
+        val oldMeta = infotonData.meta
+        val newMeta = oldMeta.copy(path = transform(transformations, oldMeta.path))
+        val infotonQuads = infotonData.data.utf8String.split('\n')
+        val newData = infotonQuads.foldLeft(StringBuilder.newBuilder) { (total, line) =>
+          val subjectEndPos = line.indexOf(' ')
+          val predicateEndPos = line.indexOf(' ', subjectEndPos + 1)
+          val isValueReference = line.charAt(predicateEndPos + 1) == '<'
+          val lastSpaceBeforeLastPart = line.lastIndexOf(' ', line.length - 3)
+          val isLastReference = line.charAt(line.length - 3) == '>' && !line.substring(lastSpaceBeforeLastPart).contains("^^")
+          val isQuad = isLastReference && lastSpaceBeforeLastPart != predicateEndPos
+          val valueEndPos = if (isQuad) lastSpaceBeforeLastPart else line.length - 2
+          val newSubject = transform(transformations, line.substring(0, subjectEndPos + 1))
+          val predicate = line.substring(subjectEndPos + 1, predicateEndPos + 1)
+          val oldValue = line.substring(predicateEndPos + 1, valueEndPos + 1)
+          val newValue = if (isValueReference) transform(transformations, oldValue) else oldValue
+          val newQuad = if (isQuad) transform(transformations, line.substring(valueEndPos + 1)) else line.substring(valueEndPos + 1)
```

Comment:
Why is that needed? 
If not quad -> it's a value and you already took care of it